lib/pymedphys/_streamlit/utilities/mosaiq.py

```diff
@@ -14,7 +14,9 @@
 
 from pymedphys._imports import streamlit as st
 
+from pymedphys._mosaiq import _separate_server_port_string
```

Comment:
I don't know if importing _separate_server_port_string is a great idea, but it seemed the least intrusive approach

Reply:
Yeah, I'd recommend renaming that function from `_separate_server_port_string` to `separate_server_port_string`, add a short docstring, and then utilise it within `_streamlit/utilities/mosaiq` as you have. One thing though, instead of re-importing, just utilise the already imported `msq_connect` scope:

```python
  hostname, port = msq_connect.separate_server_port_string(server)  # probably best not reusing server variable here
  storage_name = msq_connect.get_storage_name(hostname, port)
  username, password = msq_connect.get_username_and_password_without_prompt(
      storage_name
  )
```

Also, after renaming something like that, it can be really helpful to run:

```
poetry run pymedphys dev lint
```

It'll tell you all the places in the code base where that name change was problematic.

Reply:
Or, another option (up to you), might be to create a function within `_mosaiq.connect` that looks like the following:

```python
def get_username_and_password_from_hostname_port_string(hostname_port_string)
	  hostname, port = separate_server_port_string(hostname_port_string)
	  storage_name = get_storage_name(hostname, port)
	  username, password = get_username_and_password_without_prompt(
	      storage_name
	  )
```